Mono.Debugging.Soft/SoftDebuggerSession.cs

```diff
@@ -737,19 +739,84 @@ protected override ProcessInfo[] OnGetProcesses ()
 		internal PortablePdbData GetPdbData (AssemblyMirror asm)
 		{
 			string assemblyFileName;
+
 			if (!assemblyPathMap.TryGetValue (asm.GetName ().FullName, out assemblyFileName))
 				assemblyFileName = asm.Location;
 			var pdbFileName = Path.ChangeExtension (assemblyFileName, ".pdb");
-			if (!PortablePdbData.IsPortablePdb (pdbFileName))
-				return null;
-			return new PortablePdbData (pdbFileName);
+			if (PortablePdbData.IsPortablePdb (pdbFileName))
+				return new PortablePdbData (pdbFileName);
+			// Attempt to fetch pdb from the debuggee over the wire
+			var pdbBlob = asm.GetPdbBlob ();
+			return pdbBlob != null ? new PortablePdbData (pdbBlob) : null;
 		}
 
 		internal PortablePdbData GetPdbData (MethodMirror method)
 		{
 			return GetPdbData (method.DeclaringType.Assembly);
 		}
 
+		Dictionary<AssemblyMirror, SourceLinkMap []> SourceLinkCache = new Dictionary<AssemblyMirror, SourceLinkMap []> ();
+		SourceLinkMap [] GetSourceLinkMaps (MethodMirror method)
+		{
+			var asm = method.DeclaringType.Assembly;
+			SourceLinkMap [] maps = Array.Empty<SourceLinkMap>();
+
+			if (SourceLinkCache.TryGetValue (asm, out maps)) {
```

Comment:
```suggestion
			if (SourceLinkCache.TryGetValue (asm, out SourceLinkMap[] maps)) {
```

Reply:
Combined with the previous suggestion, maps can now be null? I was ensuring that it could never be null.

Reply:
Any out var is going to be assigned by the caller. So regardless of whatever is before it, it is going to be assigned (most likely to the default) by the callee.

Reply:
Yep, I see now. Fixed